lib/authentication/view/edit_profile_page.dart

```diff
@@ -22,10 +24,94 @@ class EditProfilePage extends StatefulWidget {
 class _EditProfilePageState extends State<EditProfilePage> {
   final firstNameController = TextEditingController();
   final lastNameController = TextEditingController();
+  final emailController = TextEditingController();
   final dropdownController = DropdownTreeController();
 
   final formKey = GlobalKey<FormState>();
 
+  AppDialog _changePasswordDialog(BuildContext context) {
+    final newPasswordController = TextEditingController();
+    final oldPasswordController = TextEditingController();
+    final changePasswordKey = GlobalKey<FormState>();
```

Comment:
I just checked the UI, and the mixed use of form validation with toast errors is ugly: if I don't provide a password, I get a red error under the text field. If I provide a wrong password or one that's too short, I get a toast. That's bad inconsistency.

Pick either one or the other and stick with it. I recommend form validation and use the toast only for the success message.

Reply:
Toast is better, it is consistent with the sing up page. May be in a future PR change all of them to validation?

Reply:
The login/signup UI is completely separate from everything else, both because it was the first thing I did and I didn't really know what I was doing, and because it gives a feel of "I'm outside of the app". Every form **inside** the app uses form validation over toasts.

Reply:
Not to mention, the signup form has a completely different UI for showing errors, which you don't use here (the x at the end that you can press to show the error). So really, using toasts here is inconsistent with everything.